File: tests/rspec/pages/files/file_manager.rb
class FileManager < FileManagerPage
  set_url_matcher /files/

  # Main box elements
  element :heading, 'div.col.w-12 div.box form h1'
  element :sync_button, 'a.icon--sync'

  element :perpage_filter, 'div.col.w-12 div.box form h1 + div.filters ul li:nth-child(2)'
  element :perpage_filter_menu, 'div.col.w-12 div.box form h1 + div.filters ul li:nth-child(2) div.sub-menu ul'
  element :perpage_manual_filter, 'input[name="perpage"]'

  # Main box's table elements
  elements :files, 'div.box form div.tbl-wrap table tr'
  element :selected_file, 'div.box form div.tbl-wrap table tr.selected'

  element :title_name_header, 'div.box form div.tbl-wrap table tr th:first-child'
  element :file_type_header, 'div.box form div.tbl-wrap table tr th:nth-child(2)'
  element :date_added_header, 'div.box form div.tbl-wrap table tr th:nth-child(3)'
  element :manage_header, 'div.box form div.tbl-wrap table tr th:nth-child(4)'
  element :checkbox_header, 'div.box form div.tbl-wrap table tr th:nth-child(5)'

  elements :title_names, 'div.box form div.tbl-wrap table tr td:first-child'
  elements :file_types, 'div.box form div.tbl-wrap table tr td:nth-child(2)'
  elements :dates_added, 'div.box form div.tbl-wrap table tr td:nth-child(3)'
  elements :manage_actions, 'div.box form div.tbl-wrap table tr td:nth-child(4)'

  element :bulk_action, 'form fieldset.tbl-bulk-act select[name="bulk_action"]'
  element :action_submit_button, 'form fieldset.tbl-bulk-act button.submit'

  element :no_results, 'tr.no-results'

  element :view_modal, 'div.modal-view-file'
  element :view_modal_header, 'div.modal-view-file h1'

  element :remove_directory_modal, 'div.modal-confirm-directory'
  element :remove_directory_modal_submit_button, 'div.modal-confirm-directory .form-ctrls input.btn'

  def load
    click_link 'Files'
  end

end

File: tests/cypress/cypress/elements/pages/files/FileManager.js
import FileManagerSection from '../_sections/FileManagerSection'

class FileManager extends FileManagerSection {
    constructor() {
        super()
				this.url = 'admin.php?/cp/files';
				this.urlMatch = /files/;

        this.elements({
					// Main box elements
					'heading': '.ee-main .title-bar .title-bar__title',
					'sync_button': 'a.icon--sync',

					'perpage_filter': '.pagination__item .filter-search-bar__item',
					'perpage_filter_menu': '.pagination__item .filter-bar__button[data-filter-label="show"] + .dropdown',
					'perpage_manual_filter': '.pagination__item input[type!=hidden][name="perpage"]',

					// Main box's table elements
					'files': '.ee-main__content form .table-responsive table tbody tr',


					'title_name_header': '.ee-main__content form .table-responsive table tr th:nth-child(3)',
					'file_type_header': '.ee-main__content form .table-responsive table tr th:nth-child(5)',
					'date_added_header': '.ee-main__content form .table-responsive table tr th:nth-child(6)',
					'manage_header': '.ee-main__content form .table-responsive table tr th:nth-child(8)',
					'checkbox_header': '.ee-main__content form .table-responsive table tr th:nth-child(1) input[type=checkbox]',

					'title_names': '.ee-main__content form .table-responsive table tr td:nth-child(3)',
					'file_types': '.ee-main__content form .table-responsive table tr td:nth-child(5)',
					'dates_added': '.ee-main__content form .table-responsive table tr td:nth-child(6)',
					'manage_actions': '.ee-main__content form .table-responsive table tr td:nth-child(8)',

					'no_results': '.no-results',

					'view_modal': 'div.modal-view-file',
					'view_modal_header': 'div.modal-view-file h1',

					'remove_directory_modal': 'div.modal-confirm-directory',
					'remove_directory_modal_submit_button': 'div.modal-confirm-directory .form-ctrls .button',

                    'new_folder_button': '.title-bar .btn.new_folder',

                    'file_input': 'div[data-input-value="files_field"]',
                    
        })
		}
		load() {
			cy.contains('Files').click()
		}
        load_for_dir(number) {
            cy.contains('Files').click()
            cy.get('div.sidebar .folder-list > div:nth-child('+number.toString()+')').click();
        }
}
export default FileManager;
